### ISSUE TITLE
Bring back Scaling WITHOUT respecting the remote destkop aspect ratio

### ISSUE THREAD

**Issue by Lonniebiz** [FreeRDP/Remmina#537]
Previous versions of Remmina were capable of scaling **without** respecting aspect ratio. The newest version has removed this feature so that now Remmina only scales with strict adherence to the aspect ratio.

Believe it or not, **not** respecting aspect ratio was one of my favorite features in Remmina. While one might say it distorts the desktop, I say it magnifies it and my eyes just love it.

When I do a remote rdp, I set the resolution 1024x768, then I scale it without respecting aspect ratio. This make the remote desktop utilize the entire size of my viewport. Yes, it does "distort" things to look a little wider than they normally would, but I absolutely love it.

Look at this screenshot of how things are now in the newest Reminna:
[http://neartalk.com/ss/2015-04-06_001_1916x1078.png](http://neartalk.com/ss/2015-04-06_001_1916x1078.png)

This is scaled, but look how much black-space I have on both sides of the remote desktop. Before, it would fill up the entire tab no black space on any side. Yes, not respecting aspect ratio distorts (making things look a little wider), but this is quite favorable to me.

I've been using Remmina for a long time now, and I've come to depend on this feature. When you (instead) maintain aspect ratio, it makes me feel like I'm trapped in a little box. I know not everyone is like me, but I like things to be big while I'm working. I've been straining my eyes on these computers for 20 years, and Remmina has been a **blessing** to my aging eyes. I use it as a tool to MAGNIFY all these remote servers and workstation-desktops I work on each day. Most of the work I'm doing doesn't require me to see a huge workspaces (I don't need the room), so I just prefer that everything be huge and because of this I hardly ever strain my eyes anymore.

Please bring back this feature. If not by default, please allow me to set it as my default preference in the settings menu.

**Comment by Lonniebiz** [comment]
For your convenience, I've just downgraded from Ubuntu 15.04 to Ubuntu 14.04 so that I can show you a comparison.

Previous, I showed a desktop scaled while maintaining aspect ratio (from the newest version of Remmina: 
http://neartalk.com/ss/2015-04-06_001_1916x1078.png

Here's that same desktop in the Remmina that comes with Ubuntu 14.04, scaled **without** maintaining aspect ratio:
http://neartalk.com/ss/2015-04-06_001_1919x1079.jpg

With this side by side comparison, hopefully you can see why one might want what I'm requesting. Yes, there is some distortion, but it is worth it for me to fill up my viewport. Not everyone will agree, but that's what preferences are for.

**Comment by Lonniebiz** [comment]
I just updated that second screenshot, it was distorted due to compression in Shutter (not by Remmina). So if you viewed it in email, you might want to take a second look:
http://neartalk.com/ss/2015-04-06_002_1918x1079.png

**Comment by giox069** [comment]
![2015-04-06-213858_1920x1080_scrot](https://cloud.githubusercontent.com/assets/5477803/7010983/a998f32a-dca5-11e4-9089-b811e7da425f.png)
I just added a scaler option to expand the remote desktop without maintaining its aspect ratio. If you are using ubuntu, you should receive an update via the PPA in 24 hours.
Test it and let us know.

**Comment by Lonniebiz** [comment]
Thank you for addressing this so quickly.

I'm currently running Ubuntu 14.04 with the intent of moving to 15.04 if this feature you've added can be persisted in my use of Remmina.

Anyway, I don't think I'm testing the right thing. Here's what I've just done:

sudo apt-add-repository ppa:remmina-ppa-team/remmina-next
sudo apt-get update
sudo apt-get install remmina remmina-plugin-rdp

I think this is the wrong ppa to test what you've done. What's the right one?

**Comment by giox069** [comment]
remmina-master PPA is the old version of remmina (1.1.X), and the sliders you are seeing comes from the old scaler code, now removed. Please do not use it.

remmina-next PPA has the new version of remmina, please use remmina-next and check the version of remmina in the Help->About menu: the current version of remmina we are working on is named "1.2.0-rcgit.3".
Also as reference check the page of the [remmina-next PPA](https://launchpad.net/~remmina-ppa-team/+archive/ubuntu/remmina-next): the pakage names 1.1.1...next.... are correct. They are named 1.1.1 but they contains Remmina 1.2.0-rcgit.X.

**Comment by Lonniebiz** [comment]
So, I uninstalled remmina, and then remove the master ppa.

After adding the remmina-next ppa, I install remmina again, and it says it is version 1.1.2.

Here's the ppa I'm using:
http://neartalk.com/ss/2015-04-10_002_1034x65.png

But after reinstalling, Remmina is not 1.2:
http://neartalk.com/ss/2015-04-10_001_425x402.png

Maybe upgrading from Ubuntu 14.04 to 15.04 will make the remmina-next ppa provide 1.2 ?

I read what you said about how it is named 1.1.1, but is actually 1.2, but I still see the sliders and chain link regarding aspect ratio, not your additions.

**Comment by giox069** [comment]
I just installed a fresh virtual machine with Ubuntu 14.04.2, added the remmina-ppa-team/remmina-next ppa, and installed: remmina is the correct 1.2.0-rcgit.3 version.

So you must have something not working in your system.

1. Look into directory ``/etc/apt/soruces.list.d/`` and remove all remmina related PPA, then do an ``apt-get update``.
2. Remove all remmina and freerdp packages, I hope this command will do enough:
``sudo apt-get --purge remove freerdp-x11
 remmina remmina-common remmina-plugin-rdp remmina-plugin-vnc remmina-plugin-gnome
 remmina-plugin-nx remmina-plugin-telepathy remmina-plugin-xdmcp``
3. Check that no other remmina and freerdp packages are installed
``dpkg --list | egrep "(remmina|freerdp)"``
If the above command shows some installed package, remove it with your favorite package manager (i.e.: apt-get remove *packagename*)
4. Check that no other remmina executables are left on your executable PATH:
``hash -r``
an then
``which remmina``
should give no output. Also the command ``which freerdp`` should give no output
5. Reinstall the remmina-next PPA as explained at the top of the [Remmina Wiki](https://github.com/FreeRDP/Remmina/wiki)

**Comment by Lonniebiz** [comment]
I throughly followed your instructions. Step 4 checked out fine returning nothing. Then after step 5:

Here's what is installed:
http://neartalk.com/ss/2015-04-10_003_1547x876.png
http://neartalk.com/ss/2015-04-10_006_1547x876.png

Here's how Remmina looks:
http://neartalk.com/ss/2015-04-10_005_1066x600.png

**Comment by giox069** [comment]
Packages names seems to be ok. Can you open a terminal and type
``which remmina`` and post the output here ?
Also please try to launch remmina from the terminal by typing ``killall remmina && remmina`` and pressing enter, and then check the version.

**Comment by Lonniebiz** [comment]
So, it seems there is a step 6: Reboot Your Computer :)

The last steps you mention may have worked too, but I've just log onto this computer again, and I now I see the new Remmina; Ubuntu somehow held onto the old version until reboot.

**Comment by Lonniebiz** [comment]
Ok, the feature you added works great!

There is one problem though, I have to repeat the steps of selecting this preference:
1) Each time I launch Remmina
2) Each time I launch a new remote session from an already launched Remmina

The steps I have to repeat each time are:
1) Click the "Scale the desktop" button
2) Click the "Scaling options" button
3) Select the "Fill client when scaled" option (you've added)

This might not seem like much of a burden, and it isn't for people who launch one remote session in a day. However, if you launch 50 in a day, that's 150 additional steps.

I've always wished that Remmina would persist my preferences so that both 1) "Between sessions" and 2) "during sessions" it would always persist my preference of launching new remotes sessions as "scaled without respect to aspect ratio". Otherwise, I have to repeat these step for each remote desktop I launch (several times per day).

Even in the earlier versions, I've always had to click the "Scale" button for each session I launch. Now, I have to do that and also select the feature you've added (two additional clicks per remote session).

Look at the preferences dialog:
http://neartalk.com/ss/2015-04-10_007_813x531.png

I have Scrolled Window selected, but there is no option for:
"Full Size in Tab and Scaled Without Respect to Aspect Ratio"

**Comment by giox069** [comment]
**ABOUT step 6:**
You are right :) Remmina has not an option to exit immediately, and it will be running forever. This is why step 6 is missing. Step 6 could be one of the following.
1. reboot computer (as you said)
2. close the remmina main window, and then go on the appindicator/systemtray remmina applet menu, and then choose "exit". Wait 10 seconds, and remmina should end.
3. kill the process remmina, via your favorite process killing tool. For example on a terminal window, type "killall remmina".
**About scaling preferences not saved**
Scaling preferences are saved automatically every time you close a remote connection, if you are using a saved profile. They are not saved when you use the "Quick connect" button.

I agree with you that in the global preferences dialog there is not an option to choose the default exapand method, and maybe we can add it in the future. I will reopen another issue in the next days.

**Comment by Lonniebiz** [comment]
Do you have a pay pal account? How can I send you a donation? Also, I how do I donate to the Remmina project in General? I saw no paypal account on the project page.

**Comment by giox069** [comment]
Thank you for offering a donation, we appreciate it, but currently we don't accept donations, and I think we will not accept donations for the next months. There is an old donation page on the old sourceforge remmina site which points to the old remmina author and copyright holder paypal account. But a donation to him will not be useful for the current project maintainers (only 2 peoples, unfortunately), because he no longer maintains remmina.
